app/javascript/packages/phone-input/index.js

```diff
@@ -42,15 +42,20 @@ export class PhoneInput extends HTMLElement {
   /** @type {string[]} */
   deliveryMethods = [];
 
+  /** @type {Object.<string,*>} */
```

Comment:
It looks to me like it works in practice, though my understanding is that [it's recommended to avoid the uppercase "Object" type (along with a few others)](https://www.typescriptlang.org/docs/handbook/declaration-files/do-s-and-don-ts.html#number-string-boolean-symbol-and-object). Also, would we be safe to assume that the expected value should be a string?

Maybe one of these:

```suggestion
  /** @type {Record<string, string>} */
```

```suggestion
  /** @type {object} */
```

The latter comes from [the DefinitelyTyped types for the library](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/71d3f4939ce39b9be3e177bde32c1b77d9538115/types/intl-tel-input/index.d.ts#L222), in case we want to align to that, and/or add it as a devDependency to our project.